blog/2025/pcsx2-2.6/index.mdx

```diff
@@ -0,0 +1,606 @@
+
+Some optimizations are missing on Direct3D12 like [#13525](https://github.com/PCSX2/pcsx2/pull/13525) but will be implemented during the v2.7 dev cycle.
+
+<Image src={require("./img/benches/13549-Enth-pt1.webp").default} />
+
+<PCSX2PRLink authors="TJnotJT,refractionpcsx2" prNums="13142,13138,13005,13685">
+  ### GS/HW: Shuffle improvements
+</PCSX2PRLink>
+
+Several fixes have been made to texture shuffle emulation in order to fix some long-standing games.
+
+#### Handles Texture Shuffle With Pixel Reversals ([#13142](https://github.com/PCSX2/pcsx2/pull/13142))
+
+Detect when a game (Colin McRae Rally 2005) is doing a texture shuffle and (a) flipping pixel locations horizontally in columns and (b) correcting the pixel locations by flipping pixels horizontally again. Ignore both flips so that they cancel out. To keep it short what this does is fix Colin McRae Rally and is now playable with no hardware rendering issues.
+
+<SliderCompare
+  before={require("./img/comparisons/13142-CMR-before.webp").default}
+  after={require("./img/comparisons/13142-CMR-after.webp").default}
+/>
+
+#### Handles Texture Shuffle Sprite Not 16-pixel-multiple Wide ([#13138](https://github.com/PCSX2/pcsx2/pull/13138))
+
+Do special handling for texture shuffling sprites that are not 8 pixels wide or a multiple of 16 pixels wide. Align such sprites to 16 pixel boundaries horizontally. This fixes some issues on Alter Echo.
+
+#### Add CRC for Sand Grain Games Palette Shuffle Effect ([#13005](https://github.com/PCSX2/pcsx2/pull/13005))
+
+We have no way to do this effect directly without having 50-100 readbacks per frame, which is very very slow, this emulates the effect they're trying to achieve (G->A shuffle with palette).
+
+This fixes some long standing issues with Cabela's games, image comparisons below:
+
+<SliderCompare
+  before={require("./img/comparisons/13005-Cabela-before.webp").default}
+  after={require("./img/comparisons/13005-Cabela-after.webp").default}
+/>
+
+<SliderCompare
+  before={require("./img/comparisons/13005-CabelaLegend-before.webp").default}
+  after={require("./img/comparisons/13005-CabelaLegend-after.webp").default}
+/>
+
+#### Detect offset shuffle on TexIsFB ([#13685](https://github.com/PCSX2/pcsx2/pull/13685))
+
+Updates TexIsFB to check for horizontally offset shuffle, this fixes a shuffle issue in Timesplitters 2 where it copied the wrong data.
+
+<PCSX2PRLink authors="TJnotJT" prNums="13240">
+  ### GS/SW: Do Pre-Clamping of ST Coordinates When They May Overflow
+</PCSX2PRLink>
+
+The input texture ST coordinates can sometimes be huge or NaN (for reasons not totally clear, it might be emulation or game bugs) when in rasterizer so they cause issues with calculating the vertex trace values, determining texture bounding boxes, etc. This can cause graphical glitches downstream. This PR attempts to address such issues doing the following:
```

Comment:
```suggestion
The input texture ST coordinates can sometimes be extremely large or NaN, for reasons that aren’t fully clear (it could be due to emulation quirks or game bugs). When these coordinates reach the rasterizer, they can cause issues with calculating vertex trace values, determining texture bounding boxes, and other related computations. These problems may lead to graphical glitches downstream. This PR aims to address these issues by applying the following fixes:
```